### PR TITLE
added repository, service, controller profile

### PR DESCRIPTION
**Pull request by mariyazaiceva**

### CODE REVIEW

File: src/main/java/internship/mbicycle/storify/controller/ProfileController.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequiredArgsConstructor
+public class ProfileController {
+
+    private final ProfileService profileService;
+
+    @GetMapping("/profile/{id}")
```
Comment (Estrin-MBicycle) [inline src/main/java/internship/mbicycle/storify/controller/ProfileController.java:14]: Bad endpoint naming naming

File: src/main/java/internship/mbicycle/storify/controller/ProfileController.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package internship.mbicycle.storify.controller;
+
+import internship.mbicycle.storify.dto.ProfileDTO;
+import internship.mbicycle.storify.service.ProfileService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequiredArgsConstructor
+public class ProfileController {
```
Comment (Estrin-MBicycle) [inline src/main/java/internship/mbicycle/storify/controller/ProfileController.java:10]: Use @RequestMapping here

File: src/main/java/internship/mbicycle/storify/service/impl/ProfileServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class ProfileServiceImpl implements ProfileService {
+
+    private final ProfileRepository profileRepository;
+
+    @Override
+    public ProfileDTO getById(long id) {
+        Profile temp = this.profileRepository.getById(id);
```
Comment (Estrin-MBicycle) [inline src/main/java/internship/mbicycle/storify/service/impl/ProfileServiceImpl.java:20]: Use findById() to get Optional

File: src/main/java/internship/mbicycle/storify/service/impl/ProfileServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        return this.convertEntityToDTO(temp);
+    }
+
+    @Override
+    public ProfileDTO update(long id, ProfileDTO profileDTO) {
```
Comment (Estrin-MBicycle) [inline src/main/java/internship/mbicycle/storify/service/impl/ProfileServiceImpl.java:25]: update WHAT?)

**Review by Estrin-MBicycle — CHANGES_REQUESTED** [review]

File: src/main/java/internship/mbicycle/storify/controller/ProfileController.java (hunk 1 above)
Comment (mariyazaiceva) [reply src/main/java/internship/mbicycle/storify/controller/ProfileController.java]: Changed

File: src/main/java/internship/mbicycle/storify/controller/ProfileController.java (hunk 2 above)
Comment (mariyazaiceva) [reply src/main/java/internship/mbicycle/storify/controller/ProfileController.java:11]: Done

File: src/main/java/internship/mbicycle/storify/service/impl/ProfileServiceImpl.java (hunk 3 above)
Comment (mariyazaiceva) [reply src/main/java/internship/mbicycle/storify/service/impl/ProfileServiceImpl.java]: Done

File: src/main/java/internship/mbicycle/storify/service/impl/ProfileServiceImpl.java (hunk 4 above)
Comment (mariyazaiceva) [reply src/main/java/internship/mbicycle/storify/service/impl/ProfileServiceImpl.java]: Done

File: src/main/java/internship/mbicycle/storify/controller/ProfileController.java [hunk 5]
```diff
@@ -17,7 +17,7 @@ public ProfileDTO getProfileById(@PathVariable Long id) {
         return this.profileService.getById(id);
     }
 
-    @PutMapping("/update/{id}")
+    @PutMapping("/{id}")
     public ProfileDTO updateProfile(@PathVariable Long id, @RequestBody ProfileDTO profileDTO) {
         return this.profileService.updateProfile(id, profileDTO);
```
Comment (Estrin-MBicycle) [inline src/main/java/internship/mbicycle/storify/controller/ProfileController.java:22]: Why do we need 'this' ? Do you have any other profileServices unlike autowired?

**Review by Estrin-MBicycle — CHANGES_REQUESTED** [review]

File: src/main/java/internship/mbicycle/storify/controller/ProfileController.java (hunk 5 above)
Comment (mariyazaiceva) [reply src/main/java/internship/mbicycle/storify/controller/ProfileController.java:22]: No, it's habit, will fix it

**Pull request merged**